src/Limiters/limiter.jl

```diff
@@ -29,99 +29,110 @@ function quasimonotone_limiter!(
     max_ρq;
     rtol,
 )
-    space = axes(ρ)
+    if typeof(size(min_ρq)) == Tuple{Int}
+        space = axes(ρ)
+    else
+        space = axes(ρ).horizontal_space
+    end
     FT = Spaces.undertype(space)
     Nq = Spaces.Quadratures.degrees_of_freedom(Spaces.quadrature_style(space))
-    n_elems = length(min_ρq)
+    horz_n_elems = size(min_ρq, 1)
+    vert_n_elems = size(min_ρq, 2)
     slab_space = axes(Fields.slab(ρ, 1))
     q_data = Fields.field_values(zeros(slab_space))
     ρ_wJ_data = Fields.field_values(zeros(slab_space))
 
-    # Traverse elements
-    for e in 1:n_elems
-        ρ_e_slab = Fields.slab(ρ, e)
-        ρq_e_slab = Fields.slab(ρq, e)
-        ρ_e_data = Fields.field_values(ρ_e_slab)
-        ρq_e_data = Fields.field_values(ρq_e_slab)
-        # Compute ρ's and ρq's masses over an element e
-        ρ_e_mass = sum(ρ_e_slab)
-        ρq_e_mass = sum(ρq_e_slab)
-
-        # This should never happen, but if it does, don't limit
-        if ρ_e_mass <= zero(FT)
-            error("Negative elemental mass!")
-            continue
-        end
-
-        # Relax constraints to ensure limiter has a solution:
-        # This is only needed if running with the SSP CFL>1 or
-        # due to roundoff errors.
-        if ρq_e_mass < min_ρq[e] * ρ_e_mass
-            min_ρq[e] = ρq_e_mass / ρ_e_mass
-        end
-
-        if ρq_e_mass > max_ρq[e] * ρ_e_mass
-            max_ρq[e] = ρq_e_mass / ρ_e_mass
-        end
-
-        local_geometry_slab = Fields.slab(space.local_geometry, e)
+    # Traverse vertical levels
+    for v in 1:vert_n_elems
+        # On each level, traverse horizontal elements
+        for he in 1:horz_n_elems
+            ρ_e_slab = Fields.slab(ρ, v, he)
+            ρq_e_slab = Fields.slab(ρq, v, he)
+            ρ_e_data = Fields.field_values(ρ_e_slab)
+            ρq_e_data = Fields.field_values(ρq_e_slab)
+            # Compute ρ's and ρq's masses over an element he
+            ρ_e_mass = sum(ρ_e_slab)
+            ρq_e_mass = sum(ρq_e_slab)
 
-        q_data .= ρq_e_data ./ ρ_e_data
+            # This should never happen, but if it does, don't limit
+            if ρ_e_mass <= zero(FT)
+                error("Negative elemental mass!")
+                continue
```

Comment:
What's the `continue` for?-- it'll never be executed

Reply:
That's right, thanks for catching that! It was accidentally left behind. It was there before I added the `error`, as suggested in the PR's review  when this was added